Tests/Fixtures/Generated/Colors/literals-swift4/defaults-publicAccess.swift

```diff
@@ -17,12 +17,12 @@
 // swiftlint:disable identifier_name line_length type_body_length
 public extension ColorName {
   /// 0x339666ff (r: 51, g: 150, b: 102, a: 255)
-  public static let articleBody = #colorLiteral(red: 0.2, green: 0.5882353, blue: 0.4, alpha: 1.0)
+  static let articleBody = #colorLiteral(red: 0.2, green: 0.5882353, blue: 0.4, alpha: 1.0)
```

Comment:
how come this `public` suddenly disappeared here? (and same for others, `public` & `internal` seems to be removed everywhere in that PR)

Reply:
See the separate commit for this. The swift compiler in 10.2 warns about unnecessary internal/public like these.

Reply:
But won't this break things depending if they are compiled in a framework and used outside in a real project (vs when they are compiled in the same module during our compilation tests)?

Reply:
Just tried it in a playground and it works.

If you put files in the extra "Sources" folder, they're compiled as a separate module, so you need to make it public. To be extra sure, once I make the extension `internal`, it stops working.

Reply:
Ah, no, there's a case where this doesn't work:
- In a `public` extension, the `static`s work fine.
- In an `enum` inside a `public` extension, the `static` aren't visible, and each needs to be marked as public.

Reply:
Is this a behavior change specific to Swift 5, where ACLs declared at type level are now inherited that way on statics? Or was it always the case and those extra ACLs on statics were useless but what's new is they now warn about them?
(Also is this really a Swift 5 compiler warning not a SwiftLint warning?)

Reply:
This was for the Swift 4 templates, so Swift 5 compiler in Swift 4 compatibility mode. I saw the warning while running `rake output:compile`, which doesn't run any lint tasks.

Reply:
I'm just hoping then that the behavior with "Swift 5 compiler in Swift 4 mode" is the same as the behavior of the Swift 4 compiler. By design they are supposed to, but just want to be sure we don't accidentally remove visibility of our constants in Frameworks for people still using it Xcode/compiler

Reply:
Note that I tested the public/internal behaviour in playgrounds using Xcode 10.1, so Swift 4 compiler.

Reply:
Quick note, these are the warnings I'm getting with S5 in S4 mode:

> Tests/Fixtures/Generated/Colors/literals-swift4/multiple.swift:19:3: warning: 'internal' modifier is redundant for enum declared in an internal extension
>   internal enum Colors {

> Tests/Fixtures/Generated/Colors/literals-swift4/defaults-publicAccess.swift:20:3: warning: 'public' modifier is redundant for static property declared in a public extension
>   public static let articleBody = #colorLiteral(red: 0.2, green: 0.5882353, blue: 0.4, alpha: 1.0)

> Tests/Fixtures/Generated/Colors/literals-swift4/defaults.swift:20:3: warning: 'internal' modifier is redundant for static property declared in an internal extension
>   internal static let articleBody = #colorLiteral(red: 0.2, green: 0.5882353, blue: 0.4, alpha: 1.0)

> Tests/Fixtures/Generated/Colors/literals-swift4/defaults-customName.swift:19:3: warning: 'internal' modifier is redundant for static property declared in an internal > extension
  internal static let articleBody = #colorLiteral(red: 0.2, green: 0.5882353, blue: 0.4, alpha: 1.0)

Reply:
I've pushed some changes to ensure the template doesn't generate ACL specifiers inside the extension, except for deeper levels (where it's needed again). This should fix both the Xcode 10.2 warnings as well as keeping `public` working.